pkg/fsutil/fsutil_windows.go

```diff
@@ -0,0 +1,44 @@
+package fsutil
+
+import (
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"src.elv.sh/pkg/env"
+)
+
+// isExecutable determines whether pathStr refers to an executable file in PATH. It honors E:PATHEXT
+// but defaults to extensions ".com", ".exe", ".bat", ".cmd" if that env var isn't set.
+func isExecutable(pathStr string) bool {
+	var validExts []string
```

Comment:
Since the code is doing quite a bit of heavy lifting to build a slice anyway, better to build a set (`map[string]struct{}`) so that the check later can be done more easily.